Sources/ProjectSpec/Decoding.swift

```diff
@@ -4,18 +4,36 @@ import PathKit
 import Yams
 
 extension Dictionary where Key: JSONKey {
-
-    public func json<T: NamedJSONDictionaryConvertible>(atKeyPath keyPath: JSONUtilities.KeyPath, invalidItemBehaviour: InvalidItemBehaviour<T> = .remove) throws -> [T] {
+    public func json<T: NamedJSONDictionaryConvertible>(atKeyPath keyPath: JSONUtilities.KeyPath, invalidItemBehaviour: InvalidItemBehaviour<T> = .remove, parallel: Bool = false) throws -> [T] {
         guard let dictionary = json(atKeyPath: keyPath) as JSONDictionary? else {
             return []
         }
-        var items: [T] = []
-        for (key, _) in dictionary {
-            let jsonDictionary: JSONDictionary = try dictionary.json(atKeyPath: .key(key))
-            let item = try T(name: key, jsonDictionary: jsonDictionary)
-            items.append(item)
+        if parallel {
+            var itemOptionals: [T?] = Array(repeating: nil, count: dictionary.count)
+            var ops: [BlockOperation] = []
+            var idx: Int = 0
+            for (key, _) in dictionary {
+                ops.append(BlockOperation {[idx] in
+                    let jsonDictionary: JSONDictionary = try! dictionary.json(atKeyPath: .key(key))
+                    let item = try! T(name: key, jsonDictionary: jsonDictionary)
```

Comment:
no need to use exclamation mark on `try`

Reply:
agreed. any other concerns?

Reply:
nope. but bear in mind i am not a real reviewer 🙈

Reply:
According to another paring function, it would be better to set `try`, not `try!`

https://github.com/yonaskolb/XcodeGen/blob/a0e144b847c420ec6faa26fd5d029e5b94c0df95/Sources/ProjectSpec/Decoding.swift#L39

Reply:
looking at this further, i think it would be simpler to use `try!` instead of `try`. the issue is that these blocks are being run asynchronously, so they don't just bubble up to this function's caller. there are other strategies, like storing a result type in the list and then if any of the results are failures, throw the first failure as the error out of this function. we can do that if people want though

Reply:
eh, i'll just do it with `try` actually